oci-info/action/oci.py

```diff
@@ -0,0 +1,167 @@
+
+"""
+Module for interacting with OCI (Open Container Initiative) registries.
+"""
+
+from enum import Enum
+
+import httpx
+from pydantic import ValidationError
+from .oci_model import OciAnnotations, OciImageTags, OciIndex
+
+
+class OciMediaTypes(Enum):
+    """
+    Enumeration of different media types for OCI registries.
+
+    Attributes:
+        DOCKER_MANIFEST_V2_JSON (str): Media type for Docker manifest v2.
+        DOCKER_MANIFEST_LIST_V2_JSON (str): Media type for Docker manifest list v2.
+        OCI_MANIFEST_V1_JSON (str): Media type for OCI image manifest v1.
+        OCI_IMAGE_INDEX_V1_JSON (str): Media type for OCI image index v1.
+        JSON (str): Media type for JSON.
+    """
+
+    DOCKER_MANIFEST_V2_JSON = (
+        "application/vnd.docker.distribution.manifest.v2+json"
+    )
+    DOCKER_MANIFEST_LIST_V2_JSON = (
+        "application/vnd.docker.distribution.manifest.list.v2+json"
+    )
+    OCI_MANIFEST_V1_JSON = "application/vnd.oci.image.manifest.v1+json"
+    OCI_IMAGE_INDEX_V1_JSON = "application/vnd.oci.image.index.v1+json"
+    JSON = "application/json"
+
+
+class OciError(Exception):
+    """Oci class base exception"""
+
+
+class OciAnnotationsError(OciError):
+    """OciAnnotationsError exception"""
+
+
+class Oci(httpx.Client):
+    """Class for interacting with OCI (Open Container Initiative) registries."""
+
+    def __init__(
+        self,
+        user: str = "",
+        password: str = "",
+        timeout: int = 10,
+        reg_domain: str = "registry-1.docker.io",
+        reg_auth_domain: str = "auth.docker.io",
+        reg_auth_service: str = "registry.docker.io",
+        namespace: str = "library",
+    ):
+        """
+        Initialize an instance of the Oci class.
+
+        Args:
+            user (str): User for authentication.
+            password (str): Password for authentication.
+            timeout (int): Timeout in seconds for HTTP requests (default is 10).
+            reg_domain (str): Domain of the registry (default is 'registry-1.docker.io').
+            reg_auth_domain (str): Domain for authentication (default is 'auth.docker.io').
+            reg_auth_service (str): Authentication service (default is 'registry.docker.io').
+            namespace (str): Namespace within the registry (default is 'library').
+        """
+        # pylint: disable=too-many-arguments
+
+        super().__init__(
+            base_url=f"https://{reg_domain}",
+            timeout=timeout,
+            auth=(user, password) if user else None,
+            headers={"accept": ",".join([h.value for h in OciMediaTypes])},
+        )
+        self.reg_domain = reg_domain
+        self.reg_auth_domain = reg_auth_domain
+        self.reg_auth_service = reg_auth_service
+        self.namespace = namespace
+
+    def _get_data_as_dict(self, url: str) -> dict:
+        res = self.get(url)
+        res.raise_for_status()
+        return res.json()
+
+    def _get_token(self, repository) -> None:
```

Comment:
get functions and method should always return something. If not they it's not a getter.
```suggestion
    def _set_auth_token(self, repository: str) -> None:
```